pilot/pkg/networking/core/v1alpha3/listener.go

```diff
@@ -719,19 +719,19 @@ allChainsLabel:
 
 			// TODO(yxue) avoid bypassing authN using TCP
 			// Build filter chain options for listener configured with protocol sniffing
-			fcm := listener.FilterChainMatch{}
 			if chain.FilterChainMatch != nil {
-				fcm = *chain.FilterChainMatch
+				filterChainMatch = &(*chain.FilterChainMatch)
```

Comment:
why &*?

Reply:
it is actually a shallow copy as previous

Reply:
It is a little tricky, i will add some comments

Reply:
So embarrassed,  it seems this does not do a shallow copy, it still get the `chain.FilterChainMatch` address. I will fix it

Reply:
Need a temporary variable